### PR TITLE
Allow and deny list

### PR DESCRIPTION
**Pull request by uhbrar**
This implements the usage of the allow and deny list as discussed by the Operations and Workflows Working Group.

Currently, this utilizes the SmartAPI title for the service, as discussed during the last working group meeting. Before merging, it should be decided whether this is really the route to take, or whether or not it is more appropriate to use infores identifiers.

### CODE REVIEW

File: app/server.py [hunk 1]
```diff
@@ -101,8 +103,23 @@ async def run_workflow(
 
     async with httpx.AsyncClient(verify=False, timeout=60.0) as client:
         for operation in workflow:
+            operation_services = []
+            runner_parameters = operation.pop("runner_parameters", {})
+            if "allowlist" in runner_parameters.keys():
+                for service in SERVICES[operation["id"]]:
```
Comment (kennethmorton) [inline app/server.py:109]: We need a safety check here that operation["id"] is present in SERVICES. #36

File: app/server.py [hunk 2]
```diff
@@ -101,8 +103,23 @@ async def run_workflow(
...
+                    if service["id"] in runner_parameters["allowlist"]:
+                        operation_services.append(service)
+            else:
+                for service in SERVICES[operation["id"]]:
```
Comment (kennethmorton) [inline app/server.py:113]: Same here

File: app/server.py [hunk 3]
```diff
@@ -101,8 +103,23 @@ async def run_workflow(
...
+                    operation_services.append(service)
+                if "denylist" in runner_parameters.keys():
+                    for service in operation_services:
+                        if service["id"] in runner_parameters["denylist"]:
+                            operation_services.remove(service)
+            logger.debug(f"Service providers to query for operation '{operation}':'{operation_services}'")
+
             service_operation_responses = []
-            for service in SERVICES[operation["id"]]:
+            for service in operation_services:
```
Comment (kennethmorton) [inline app/server.py:122]: We should have a check prior to this line that we actually have any operation_services. If we don't we should send something helpful. See #36

File: requirements-lock.txt [hunk 4]
```diff
@@ -7,9 +7,9 @@ httpcore==0.13.7
 httpx==0.18.1
 idna==3.3
 pydantic==1.9.0
-reasoner-pydantic==2.1.0
+reasoner-pydantic==2.2.1
```
Comment (kennethmorton) [inline requirements-lock.txt:10]: A new version of reasoner-pydantic will be out very soon. We might want to wait to merge this until after upgrading.

File: requirements.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
 fastapi==0.63.0
 httpx==0.18.1
-reasoner-pydantic==2.1.0
+reasoner-pydantic==2.2.1
```
Comment (kennethmorton) [inline requirements.txt:3]: Wait for new version. See above.

**Review by kennethmorton — CHANGES_REQUESTED** [review]

**Comment by uhbrar** [comment]
I think it would be best to address these issues in a separate branch, for tracking purposes. This pull request is an enhancement. I plan on creating a separate pull request to handle bug fixes.

**Comment by kennethmorton** [comment]
Fair enough.

**Pull request merged**